core/src/test/java/io/questdb/test/cutlass/http/line/LineHttpSenderMockServerTest.java

```diff
@@ -0,0 +1,464 @@
+        MockHttpProcessor mockHttpProcessor = new MockHttpProcessor()
+                .withExpectedContent("test,sym=bol x=1.0\n")
+                .withExpectedHeader("User-Agent", "QuestDB/java/" + QUESTDB_VERSION)
+                .withExpectedHeader("Authorization", "Basic QWxhZGRpbjpPcGVuU2VzYW1l")
+                .replyWithStatus(204);
+        testWithMock(mockHttpProcessor, sender -> {
+            sender.table("test")
+                    .symbol("sym", "bol")
+                    .doubleColumn("x", 1.0)
+                    .atNow();
+        }, port -> Sender.builder().url("http://Aladdin:OpenSesame@localhost:" + port));
+    }
+
+    @Test
+    public void testJsonError() throws Exception {
+        String jsonResponse = "{\"code\": \"invalid\",\n" +
+                "                    \"message\": \"failed to parse line protocol: invalid field format\",\n" +
+                "                    \"errorId\": \"ABC-2\",\n" +
+                "                    \"line\": 2}";
+
+        MockHttpProcessor mockHttpProcessor = new MockHttpProcessor()
+                .withExpectedHeader("User-Agent", "QuestDB/java/" + QUESTDB_VERSION)
+                .replyWithContent(400, jsonResponse, HttpConstants.CONTENT_TYPE_JSON);
+
+        testWithMock(mockHttpProcessor, errorVerifier("Could not flush buffer: failed to parse line protocol: invalid field format [http-status=400, id: ABC-2, code: invalid, line: 2]"));
+    }
+
+    @Test
+    public void testMaxPendingRows() throws Exception {
+        MockHttpProcessor mockHttpProcessor = new MockHttpProcessor()
+                .withExpectedContent("test x=1.0\n")
+                .replyWithStatus(204)
+                .withExpectedContent("test x=2.0\n" +
+                        "test x=3.0\n")
+                .replyWithStatus(204)
+                .withExpectedContent("test x=4.0\n" +
+                        "test x=5.0\n")
+                .replyWithStatus(204)
+                .withExpectedContent("test x=6.0\n")
+                .replyWithStatus(204);
+
+        testWithMock(mockHttpProcessor, sender -> {
+            // first row to be flushed explicitly
+            sender.table("test").doubleColumn("x", 1.0).atNow();
+            sender.flush();
+
+            // 1st implicit batch sent due to maxPendingRows
+            sender.table("test").doubleColumn("x", 2.0).atNow();
+            sender.table("test").doubleColumn("x", 3.0).atNow();
+
+            // 2nd implicit batch sent due to maxPendingRows
+            sender.table("test").doubleColumn("x", 4.0).atNow();
+            sender.table("test").doubleColumn("x", 5.0).atNow();
+
+            // the last row is flushed on close()
+            sender.table("test").doubleColumn("x", 6.0).atNow();
+        }, DEFAULT_FACTORY.andThen(b -> b.maxPendingRows(2)));
+    }
+
+    @Test
+    public void testMaxPendingRows_doubleConfiguration() {
+        try {
+            Sender.builder().maxPendingRows(1).maxPendingRows(1);
```

Comment:
This and some of the below tests look like something that belong to `LineSenderBuilderTest`